java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -1165,65 +1257,70 @@ OLD: KE1
             callable,
             enclosingStmt
         )
+*/
+context(KaSession)
+private fun KotlinFileExtractor.extractLoop(
+    loop: KtLoopExpression,
+    bodyIdx: Int?,
+    stmtExprParent: StmtExprParent,
+    callable: Label<out DbCallable>,
+    getId: (Label<out DbStmtparent>, Int) -> Label<out DbStmt>
+): Label<out DbStmt> {
+    val stmtParent = stmtExprParent.stmt(loop, callable)
+    val locId = tw.getLocation(loop)
+
+    val idx: Int
+    val parent: Label<out DbStmtparent>
+
+    val label = (loop.parent as? KtLabeledExpression)?.getLabelName()
+    if (label != null) {
+        val labeledStmt = tw.getFreshIdLabel<DbLabeledstmt>()
+        tw.writeStmts_labeledstmt(labeledStmt, stmtParent.parent, stmtParent.idx, callable)
+        tw.writeHasLocation(labeledStmt, locId)
+
+        tw.writeNamestrings(label, "", labeledStmt)
+        idx = 0
+        parent = labeledStmt
+    } else {
+        idx = stmtParent.idx
+        parent = stmtParent.parent
+    }
 
-    private fun extractLoop(
-        loop: IrLoop,
-        bodyIdx: Int?,
-        stmtExprParent: StmtExprParent,
-        callable: Label<out DbCallable>,
-        getId: (Label<out DbStmtparent>, Int) -> Label<out DbStmt>
-    ): Label<out DbStmt> {
-        val stmtParent = stmtExprParent.stmt(loop, callable)
-        val locId = tw.getLocation(loop)
-
-        val idx: Int
-        val parent: Label<out DbStmtparent>
-
-        val label = loop.label
-        if (label != null) {
-            val labeledStmt = tw.getFreshIdLabel<DbLabeledstmt>()
-            tw.writeStmts_labeledstmt(labeledStmt, stmtParent.parent, stmtParent.idx, callable)
-            tw.writeHasLocation(labeledStmt, locId)
-
-            tw.writeNamestrings(label, "", labeledStmt)
-            idx = 0
-            parent = labeledStmt
-        } else {
-            idx = stmtParent.idx
-            parent = stmtParent.parent
-        }
-
-        val id = getId(parent, idx)
-        tw.writeHasLocation(id, locId)
-
-        val body = loop.body
-        if (body != null && bodyIdx != null) {
-            extractExpressionStmt(body, callable, id, bodyIdx)
-        }
+    val id = getId(parent, idx)
+    tw.writeHasLocation(id, locId)
 
-        return id
+    val body = loop.body
+    if (body != null && bodyIdx != null) {
```

Comment:
Are there some theoretically impossible situations we should log about here? e.g. non-null-body with nulll-index?

Reply:
It looks like the `null` case was introduced to handle enhanced for loops.